agent/functional_tests/testdata/taskdefinitions/container-ordering/task-definition.json

```diff
@@ -0,0 +1,77 @@
+{
```

Comment:
So, I understand that this task def uses many of the fields necessary for ordering containers. However, I'm unable to understand how running this test ensures that the feature is working correctly. As in, how are we validating that `"success"` is only running when `"success-dependency"` has reached the `"SUCCESS"` state? For all we know, ECS agent could be just ignoring those fields. Shouldn't that functionality testing be part of a functional test?

If you're testing that somewhere and I'm missing that/misreading this test, please point me to the place where such validation is happening.

Reply:
We are validating that the feature is working fine as part of integ tests in ordering-integ-tests.go. Here we are just validating the happy case.

Reply:
Right, I get that. But, functional tests by definition should test the functionality of the feature that they were written for. I think the task definition should do more than just setting a field and ensuring that the task runs. For example, you could have a `"producer"` container, which creates a file in a shared task volume. The `"consumer"` container, which is ordered to start after the `"producer"` would ensure the existence of the file and that would be a better test than just doing nothing in the task def. If we're not actually testing the functionality of the feature, then there's very little value in adding these tests.

Reply:
We used similar hard-coded overrides while testing to ensure that tests wouldn't get stuck in an infinite startup cycle.

Explicitly checking order is something we have been doing in our integration tests.

Reply:
I get what you are saying. We'll make an action to add additional checks in our functional workflow before the next release. 

We will be validating this behavior manually over the next few days, too.

Reply:
> We used similar hard-coded overrides while testing to ensure that tests wouldn't get stuck in an infinite startup cycle.
> 
> Explicitly checking order is something we have been doing in our integration tests.

I'm glad that we have integration test coverage for this. But, I'm questioning functional test coverage here. Given that this (functional tests) are considered black-box tests used for testing the acceptance criteria, we can sure strive for a higher bar by having more meaningful task defs? For example, there's a reason this `"awslogs"` test tests that a log entry is available in cw logs endpoint: https://github.com/aws/amazon-ecs-agent/blob/a906f1509da9712b728b0a67f5808fc39a571228/agent/functional_tests/tests/functionaltests_unix_test.go#L318

Reply:
> I get what you are saying. We'll make an action to add additional checks in our functional workflow before the next release.

Why not this release? Why is it ok for the bar to be lower in this release?

> We will be validating this behavior manually over the next few days, too.

Again, glad that you're manually testing this. But, I think it'd be equally good/better use of resources/time to automate the same.